[18.0][MIG] l10n_es_aeat_sii_oca: Migration to v18

AaronHForgeFlow opened the pull request:
Based on https://github.com/OCA/l10n-spain/pull/3737/

Basic technical review from: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-18.0

Depends on:

- [x] https://github.com/OCA/l10n-spain/pull/4033

omar7r committed:
[MIG] l10n_es_aeat_sii: Migración a la 10.0

RamonGuiuGou committed:
[MIG] l10n_es_aeat_sii: Migration to 9.0

pedrobaeza committed:
[FIX+IMP] l10n_es_aeat_sii: Make version 9 to work

[IMP] l10n_es_aeat_sii: Change tax computation to invoice level

Odoo v9 already knows which tax generates each invoice tax line, so
we get with this:

* Avoid to make roundings that are not accurate.
* If there are manual changes in supplier invoices taxes, now we
  will be able to notify them and prevent not matching errors.
* Less difficult algorithms for getting tax amount totals.

Roodin committed:
[FIX+IMP] l10n_es_aeat_sii:

* Corrige errores y pylint
* post_init_hook
* Corregida referencia a conector
* Permisos y cmapo de posicion fiscal erróneo

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Seguridad + usabilidad

* Filtro de no enviados SII que sólo incluya 2017
* Los trabajos no realizados se eliminan en lugar de pasarlos a estado "Hecho"
  al reenviar, cancelar, etc.
* Permiso de lectura a queue.job para los gestores de la AEAT.
* Regla de seguridad para que los gestores de la AEAT sólo vean los trabajos
  del SII.
* Los trabajos se encolan como admin para evitar errores al validar por gente
  sin permisos del conector. Desafortunadamente, no se puede mantener el
  usuario original en el trabajo.

[IMP] l10n_es_aeat_sii: Share queue.job view for both types of invoices

[FIX] l10n_es_aeat_sii: Add jobs with sudo for avoiding permission errors

[FIX] l10n_es_aeat_sii: Permissions problem

No more need for connector or AEAT permissions

[IMP] l10n_es_aeat_sii: Track content sent + better layout for return

pedrorgil committed:
[IMP]l10n_es_aeat_sii: Control de respuesta 'AceptadoConErrores'

fuentes73 committed:
[IMP] l10n_es_aeat_sii: Envío de facturas alquiler de inmuebles

amcor committed:
[IMP] l10n_es_aeat_sii: Servicios intra/extra-comunitarios como no sujetos

pedrorgil committed:
[8.0][aeat_sii][FIX] Poner copy=False en invoice_jobs_ids (#585)

pedrorgil committed:
[FIX]l10n_es_aeat_sii: Corrige captura de respuesta para estado 'Parcialmente Correcto'

pedrorgil committed:
[FIX] l10n_es_aeat_sii: Fix Spanish VATs starting with N

Envío correcto para el caso de facturas de bienes o ISP emitidas a
clientes extranjeros con sucursal permanente en España.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Don't write several times on the invoice

Writing on one cursor some invoice values and on error in another,
makes the process to hang due to a lock. Deferring the write to the
end and accumulating values prevents this lock.

Closes #594

omar7r committed:
[FIX+REF] l10n_es_aeat_sii: Refactor country code get + french territories

Refactorización de la obtención del código de país y gestión correcta de
los territorios de ultramar franceses.

omar7r committed:
[FIX] l10n_es_aeat_sii: Jobs invoice + sudo in cancellation

* Se corrige la compañía con la que se crean los trabajos del conector
* Se corrige la creación del trabajo de cancelación sii

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Requeue permissions

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Correct mapping using invoices taxes

As grouped taxes are unfolded on invoice tax footer, we need to
map on SII children taxes (only one of them though), not parent ones

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Update SII technical specification link

omar7r committed:
[IMP] l10n_es_aeat_sii: Claves de registro adicionales (#607)

Se añade la posibilidad de indicar las dos caves de registro adicionales.

omar7r committed:
[FIX] l10n_es_aeat_sii: Se corrige el envío de fcaturas con ivas y no sujetos recibidas

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Remaining migration fixes

pedrorgil committed:
[8.0][IMP] l10n_es_aeat_sii: Mejor forma de calcular la fecha de registro contable para el SII (#621)

Según la conversación mantenida aquí https://groups.google.com/forum/#!searchin/openerp-spain/sii%7Csort:relevance/openerp-spain/ZM2GlTBI5qw/iOF-FF4YBQAJ , la fecha de registro contable para el SII de las facturas recibidas debe ser la fecha en que se registra la factura en el sistema informático, con independencia de la fecha que se establezca al asiento contable de la misma. Hasta ahora por defecto se establece esta fecha como la fecha del momento del envío (esto es correcto) pero se vuelve a obtener la fecha del momento del envío también para las modificaciones y cancelaciones posteriores, y esto es incorrecto ya que es necesario almacenar la fecha en que se comunica la factura por primera vez para no altararla si hay modificaciones o anulaciones posteriores. Se crea en este PR un campo para almacenar la fecha en que se registra la factura correctamente en el SII por primera vez y se utiliza este dato en los siguientes envíos de la factura al SII. Con esto se independiza completamente la fecha de registro en el SII de la fecha del asiento contable y se asegura además que nunca se comunicará fuera de plazo (8 días ahora, 4 días después) independientemente de la fecha contable que le demos al asiento. Los contables sólo deberán tener en cuenta que deben enviar las facturas que quieran deducirse en un mes antes del día 16 del mes siguiente.
He añadido además una migración para establecer el valor del nuevo campo de fecha de registro en el SII a partir del dato 'FechaRegContable' ya comunicado y que se almacena en el último envío al SII de cada factura.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Look for user current company to see if SII is enabled

In partners, the check is only visible if the company has SII enabled. Before this,
the check was done looking to the partner company, but you can leave this field
empty (specially in a multi-company environment), and then the result is not correct.

Looking to the user company gives us the proper result.

pedrorgil committed:
[IMP] l10n_es_aeat_sii: Añadido mapeo para servicios extracomunitarios

misern2 committed:
[10.0][FIX] Actualización server action a la nueva api

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Añadido control de IVA no deducible

[FIX] l10n_es_aeat_sii: Don't replace taxes in SFRSA, but add them

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: tax instead of tax_line

Cherry-picking of v8 commit doesn't serve as is as the structure is
different in >=v9.

[FIX] l10n_es_aeat_sii: Improve exemption causes in products

* Don't default to 'none'. As field is not required, there's no need for this.
* Filter out when looking for product exemption causes those equal to 'none'.

javierjcf committed:
[MIG] l10n_es_aeat_sii: Migration to 11.0

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Send inmediately if there's a previous error

jabibi committed:
[FIX] Assignment to reserved built-in symbol: type

[FIX] l10n_es_aeat_sii: call _connect_sii() using current invoice

misern2 committed:
[FIX] l10n_es_aeat_sii: Fecha Registro SII (#735)

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: remove incorrect migration scripts

omar7r committed:
[FIX] l10n_es_aeat_sii: Enviar los importes de impuesto en la moneda de la compañía (#779)

MouTio committed:
[FIX] l10n_es_aeat_sii: Replaced internal_number with move_name in sii_invoice_dict_out (#773)

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Job channel + job unlink

* On migration, the channel used for the SII jobs has not been respected, and thus
  the record rule is not correct.
* When cancelling an invoice, if you don't see the jobs (for example, an invoicing
  user) the job is not removed, and this is dangerous.

[FIX] l10n_es_aeat_sii: Adapt to changes in account_invoice_refund_link

As the module account_invoice_refund_link was overwriting several fields
and now Odoo includes an standard field for referring to the original
invoice, we use that one for the module

[IMP] l10n_es_aeat_sii: Round rectified amounts by substitution

hugosantosred committed:
[FIX] l10n_es_aeat_sii: Redondear CuotaDeducible

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Reorganize connection + better write back on cancel

* Reduce WDSL service connection code for avoiding to repeat conditions on
  each possible call, using a mapping dictionary that can be extended
  through class variable modification in _register_hook of inherited models
* Fine grain several things on the invoice vals write back on confirm invoice.
* Use the same strategy for writing back values on invoice for cancelling invoice.

[FIX] l10n_es_aeat_sii: Unicode doesn't exist in P3

[FIX] l10n_es_aeat_sii: Make cancel invoice multi-record aware

Fixes #830

jabibi committed:
[IMP] l10n_es_aeat_sii: multiple Tax Agencies (multicompany)

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Make tests inheritable + fixed dates

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Constraint not suitable for multi-company

This constraint leads to integrity errors due to missing values in companies that
doesn't have SII enabled. As the required attribute is set in view, this should be removed.

There's still the chance of writing an invoice without this key by code, but putting
a constraint seems excessive as this key is only needed when sending the invoice. If
not present, there will be an SII error that will help you to identify the problem.

angelmoya committed:
IMP SII v1.1

jabibi committed:
[IMP] SII v1.1

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Several refinements for v1.1

* Let as historic data the old agencies
* Filtered properly the exempt causes for lines with that tax
* Raise on multiple exempt causes
* Updated link to 1.1 specification

[IMP] l10n_es_aeat_sii: New README style

jabibi committed:
[FIX] l10n_es_aeat_sii: duplicated xml ID

[FIX] l10n_es_aeat_sii: getting exempt cause error

'product.product' object has no attribute 'tax_line_ids'

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Remove refund option by substituion

This refund mode (SII key 'S') is not fully working as is (it requires 3 invoices), and was
causing problems when using it. It was also understandably the default value!

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: round 'ImporteTotal'

`amount_total_company_signed` can contain unrounded amounts, so we need to round that
amount. We use python `round` for avoiding the same problem in this result.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Avoid access error to queue.job records

Putting groups in the view is not enough for preventing Odoo to load the linked one2many model,
resulting in an error for those without permissions in queue.job object. The solution for this
is to isolate that changes in a separate view with a group, and this way the fields are not
fetched anymore if the user doesn't belong to that group.

There's other view group that also has the same group (AEAT responsible), but shouldn't be
isolated, as there's a field there (sii_state), that it's needed for the visibility of
other elements, and there's no problem with them, as they are not x2x fields.

jabibi committed:
[FIX] l10n_es_aeat_sii: always sending to the test environment

pedrobaeza committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (204 of 204 strings)

Translation: l10n-spain-11.0/l10n-spain-11.0-l10n_es_aeat_sii
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-11-0/l10n-spain-11-0-l10n_es_aeat_sii/es/

[IMP] l10n_es_aeat_sii: Specific refund invoice type

Add support for specific invoice types on refunds (R1, R2 and R3).

eantones committed:
[FIX] l10n_es_aeat_sii: Permission read error when sending an invoice to SII

Introduced on commit https://github.com/OCA/l10n-spain/commit/5067c4ef2b572599f7000403113a4a39236714e9

[FIX] l10n_es_aeat_sii: float_round error in BaseImponible

Example on exempt invoices:

- "Invoice Lines" tab:
    * { quantity: 1, price_unit: 82.60, invoice_line_tax_ids: ['IVA Exento Repercutido'] }

- "SII" tab:
    - "General" tab:

        * Error de envio SII: 1100 | Valor o tipo incorrecto del campo: BaseImponible

    - "Technical" tab:

        * Último contenido enviado al SII:
        {
            "PeriodoLiquidacion": {
                "Ejercicio": 2019,
                "Periodo": "01"
            },
            "IDFactura": {
                "IDEmisorFactura": {
                    "NIF": "XXXXXXXX"
                },
                "FechaExpedicionFacturaEmisor": "07-01-2019",
                "NumSerieFacturaEmisor": "XXXXXXX"
            },
            "FacturaExpedida": {
                "TipoDesglose": {
                    "DesgloseTipoOperacion": {
                        "PrestacionServicios": {
                            "Sujeta": {
                                "Exenta": {
                                    "DetalleExenta": [
                                        {
                                            "BaseImponible": 82.60000000000001,
                                            "CausaExencion": "E1"
                                        }
                                    ]
                                }
                            }
                        }
                    }
                },
                "DescripcionOperacion": "/",
                "ImporteTotal": 82.6,
                "TipoFactura": "F2",
                "ClaveRegimenEspecialOTrascendencia": "01"
            }
        }

        * Retorno SII
        {
            'CSV': None,
            'DatosPresentacion': None,
            'Cabecera': {
                'IDVersionSii': '1.1',
                'Titular': {
                    'NombreRazon': 'XXXXXXXXXXXXXXXXXXXX',
                    'NIFRepresentante': None,
                    'NIF': 'XXXXXXXXXXX'
                },
                'TipoComunicacion': 'A0'
            },
            'EstadoEnvio': 'Incorrecto',
            'RespuestaLinea': [
                {
                    'IDFactura': {
                        'IDEmisorFactura': {
                            'NIF': 'XXXXXXXXXXX'
                        },
                        'NumSerieFacturaEmisor': 'XXXXXXXXXXXXX',
                        'NumSerieFacturaEmisorResumenFin': None,
                        'FechaExpedicionFacturaEmisor': '07-01-2019'
                    },
                    'RefExterna': None,
                    'EstadoRegistro': 'Incorrecto',
                    'CodigoErrorRegistro': 1100,
                    'DescripcionErrorRegistro': 'Valor o tipo incorrecto del campo: BaseImponible',
                    'CSV': None,
                    'RegistroDuplicado': None
                }
            ]
        }

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Update documentation to meet v11 requirements

eantones committed:
[FIX] l10n_es_aeat_sii: float_round error in BaseImponible

Extension of https://github.com/OCA/l10n-spain/pull/972/commits/8a650787f14fd40513e9161a0e78bf09b55fd220

En las lineas https://github.com/OCA/l10n-spain/blob/fabc3a39eda6c595045906a3a612a18a6d902f9f/l10n_es_aeat_sii/models/account_invoice.py#L491-L492
y
https://github.com/OCA/l10n-spain/blob/fabc3a39eda6c595045906a3a612a18a6d902f9f/l10n_es_aeat_sii/models/account_invoice.py#L537-L538

la suma (+=) genera un error de coma flotante, para reproducir el error desde la consola Python:

Python 3.5.2 (default, Nov 12 2018, 13:43:14)
[GCC 5.4.0 20160609] on linux
>>> 170.61 + 107.35
277.96000000000004

Example on customer exempt invoices:

- "Invoice Lines" tab:
    * [
        { quantity: 1,  price_unit: 56.31,  invoice_line_tax_ids: ['IVA Exento Repercutido'] },
        { quantity: 58, price_unit: 0.88,   invoice_line_tax_ids: ['IVA Exento Repercutido'] },
        { quantity: 1,  price_unit: 170.61, invoice_line_tax_ids: ['IVA Exento Repercutido'] },
      ]

- "SII" tab:
    - "General" tab:

        * Error de envio SII: 1100 | Valor o tipo incorrecto del campo: BaseImponible

    - "Technical" tab:

        * Último contenido enviado al SII
        {
            "FacturaExpedida": {
                "ClaveRegimenEspecialOTrascendencia": "01",
                "ImporteTotal": 277.96,
                "DescripcionOperacion": "/",
                "TipoDesglose": {
                    "DesgloseTipoOperacion": {
                        "PrestacionServicios": {
                            "Sujeta": {
                                "Exenta": {
                                    "DetalleExenta": [
                                        {
                                            "BaseImponible": 277.96000000000004,
                                            "CausaExencion": "E1"
                                        }
                                    ]
                                }
                            }
                        }
                    }
                },
                "TipoFactura": "F1",
                "Contraparte": {
                    "NIF": "XXXXXXXX",
                    "NombreRazon": "XXXXXXXXXXXXXXXXXXXXXXXX"
                }
            },
            "IDFactura": {
                "NumSerieFacturaEmisor": "22902",
                "FechaExpedicionFacturaEmisor": "17-01-2019",
                "IDEmisorFactura": {
                    "NIF": "XXXXXXXXXX"
                }
            },
            "PeriodoLiquidacion": {
                "Periodo": "01",
                "Ejercicio": 2019
            }
        }

        * Retorno SII

        {
            'CSV': None,
            'DatosPresentacion': None,
            'Cabecera': {
                'IDVersionSii': '1.1',
                'Titular': {
                    'NombreRazon': 'XXXXXXXXXXXX',
                    'NIFRepresentante': None,
                    'NIF': 'XXXXXXXXXXXXX'
                },
                'TipoComunicacion': 'A0'
            },
            'EstadoEnvio': 'Incorrecto',
            'RespuestaLinea': [
                {
                    'IDFactura': {
                        'IDEmisorFactura': {
                            'NIF': 'XXXXXXXXXXXX'
                        },
                        'NumSerieFacturaEmisor': '22902',
                        'NumSerieFacturaEmisorResumenFin': None,
                        'FechaExpedicionFacturaEmisor': '17-01-2019'
                    },
                    'RefExterna': None,
                    'EstadoRegistro': 'Incorrecto',
                    'CodigoErrorRegistro': 1100,
                    'DescripcionErrorRegistro': 'Valor o tipo incorrecto del campo: BaseImponible',
                    'CSV': None,
                    'RegistroDuplicado': None
                }
            ]
        }

oihane committed:
[IMP] l10n_es_aeat_sii: update wsdl links

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Direct dependency to `l10n_es`

As there's no indirect dependency.

misern2 committed:
[MIG] l10n_es_aeat_sii: Migration to 12.0

luc-demeyer committed:
[IMP] l10n_es_aeat_sii: make account_invoice sii_refund_type default inheritable

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Remove first semester of 2017 check

It's not needed anymore as we are far from that date now.

Commit:
[IMP] l10n_es_aeat_sii: No enviar al SII el mismo contenido (#892)

dalonsod committed:
[11.0][FIX] l10n_es_aeat_sii: on send and cancel error management exception object is not properly cast to string

fuentes73 committed:
[FIX] l10n_es_aeat_sii: queue_job wrong eta value

danielduqma committed:
[IMP] l10n_es_aeat_sii: Default refund specific type

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_aeat_sii
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_aeat_sii/

omar7r committed:
[FIX] l10n_es_aeat_sii: Proper check of supplier invoice number

Field name has changed.

josepmy committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (211 of 211 strings)

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_aeat_sii
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_aeat_sii/es/

eantones committed:
[FIX] l10n_es_aeat_sii: float_round error in inbound invoices

Extension of https://github.com/OCA/l10n-spain/pull/985/commits/e63a80fde8b953d09752542eece24451017a065d

Error de redondeo en las facturas entrantes

Example on vendor invoice:

- "Invoice Lines" tab:
    * [
        { quantity: 14,  price_unit: 78.48,  invoice_line_tax_ids: ['IVA Soportado no deducible 21%'] },
      ]

- "SII" tab:
    - "General" tab:

        * Error de envio SII: 1100 | Valor o tipo incorrecto del campo: CuotaSoportada

    - "Technical" tab:

        * Último contenido enviado al SII

        {
            "IDFactura": {
                "IDEmisorFactura": {
                    "NIF": "XXXXXXX"
                },
                "NumSerieFacturaEmisor": "XXXXXX",
                "FechaExpedicionFacturaEmisor": "08-08-2019"
            },
            "PeriodoLiquidacion": {
                "Ejercicio": 2019,
                "Periodo": "08"
            },
            "FacturaRecibida": {
                "TipoFactura": "F1",
                "ClaveRegimenEspecialOTrascendencia": "01",
                "DescripcionOperacion": "/",
                "DesgloseFactura": {
                    "DesgloseIVA": {
                        "DetalleIVA": [
                            {
                                "TipoImpositivo": "21.0",
                                "BaseImponible": 1098.72,
                                "CuotaSoportada": 230.73000000000002
                            }
                        ]
                    }
                },
                "Contraparte": {
                    "NombreRazon": "XXXXXXXXXXXXXX",
                    "NIF": "XXXXXXXXXXXXX"
                },
                "FechaRegContable": "08-08-2019",
                "ImporteTotal": 1329.45,
                "CuotaDeducible": 0.0
            }
        }

        * Retorno SII

        {
            'CSV': None,
            'DatosPresentacion': None,
            'Cabecera': {
                'IDVersionSii': '1.1',
                'Titular': {
                    'NombreRazon': 'XXXXXXXXXXXXXXXXX',
                    'NIFRepresentante': None,
                    'NIF': 'XXXXXXXXXXXXX'
                },
                'TipoComunicacion': 'A1'
            },
            'EstadoEnvio': 'Incorrecto',
            'RespuestaLinea': [
                {
                    'IDFactura': {
                        'IDEmisorFactura': {
                            'NIF': 'XXXXXXXXXXX',
                            'IDOtro': None
                        },
                        'NumSerieFacturaEmisor': 'XXXXXXXXXX',
                        'NumSerieFacturaEmisorResumenFin': None,
                        'FechaExpedicionFacturaEmisor': '08-08-2019'
                    },
                    'RefExterna': None,
                    'EstadoRegistro': 'Incorrecto',
                    'CodigoErrorRegistro': 1100,
                    'DescripcionErrorRegistro': 'Valor o tipo incorrecto del campo: CuotaSoportada',
                    'CSV': None,
                    'RegistroDuplicado': None
                }
            ]
        }

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Apply dict rounding at other level

Doing it at taxes dictionary level, there are some values left that are not
rounded, like 'CuotaDeducible'. We apply the rounding over the top dictionary,
also avoiding to duplicate some code.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Send now button in black color

The assigned class avoided to be shown as it appears as white.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Set refund type on direct refund creation/write

When no context is set and a refund is created, the refund type was not set. Example:

* Create a sales order with stockable product.
* Deliver it and invoice the sales order.
* Return the delivery with refund option checked.
* Invoice the refund.
* No refund type set.

With this patch, we assure that the type is set no matter the context.

MiquelRForgeFlow committed:
[UPD] Remove deprecated from_string()

MiquelRForgeFlow committed:
[UPD] Remove deprecated decodestring and encodestring

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Allow new invoice state `in_payment`

That state is not covered by the code.

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Don't depend on tax description

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_aeat_sii
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_aeat_sii/

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Include newly created non deductible taxes in mapping

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Make VAT not subject to work

These were not working since https://github.com/odoo/odoo/commit/5b61898a,
so this fixes this.

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_aeat_sii
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_aeat_sii/

angelmoya committed:
Update account_invoice.py

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Include in ROADMAP #1171

Closes #1171

JuanjoA committed:
Extract method to obtain if an invoice should be simplified or not

JuanjoA committed:
sii out_refund invoice with R1 by default

JuanjoA committed:
[FIX] l10n_es_aeat_sii: Envío al SII ImporteTotal sin IRPF en facturas de proveedor

- Añadido aeat.sii.map.lines para NotIncludedInTotal
- facturas emitidas irpf

ValentinVinagre committed:
[MIG] l10n_es_aeat_sii: Migration to 13.0

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Fix re-sending method

Previous method was comparing 2 strings for equality, but coming from dictionaries,
the generated strings may be different, as the order is not guaranteed in
<= Python 3.5.

We can use Python dictionary comparison instead for accurated results.

The method name has also been changed for reflecting the real semantics of it.

TT25025

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: VAT not subject for goods

Since odoo/odoo#34044, VAT not subject for goods was not included in any SII taxes
map, which provokes that any vendor bill of this type is not correctly sent.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Compute currency taxes amounts with proper date

If there's an accounting date, taxes amounts in company currency are not properly
computed.

The method has been refactored as well for using the modern `_convert` method.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Get default invoice type properly

You can have default invoice type expressed in the context key `default_type`, so
we have to check that as well.

eantones committed:
[FIX] l10n_es_aeat_sii: Added rounding for "Régimen agrario" key ImporteCompensacionREAGYP

danielduqma committed:
[FIX] l10n_es_aeat_sii: Default refund specific type

Don't assign a fixed refund specific type for avoiding problems in refunds of
simplified invoices

pedrobaeza committed:
[REF] l10n_es_aeat_sii: Refactor _get_sii_in_taxes for less code

This refactoring is made for reducing code and do it even more readable, but
esentially for a later improvement that will come for grouping taxes with same
percentage.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Merge tax dicts for same TipoImpositivo

If not merged, we may rebase the limit of 6 maximum blocks of this type. This can
happen when using several types of VAT taxes (goods/services) in the same invoice,
or not deductible VAT with different analytic accounts.

TT25683

pedrobaeza committed:
[FIX+IMP] l10n_es_aeat_sii: Adjust "total without IRPF" algorithm

- Don't process again everything, but do it inline on existing methods
- Make tests resilient

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: black, isort, prettier

pedrobaeza committed:
[MIG+IMP] l10n_es_aeat_sii: Complete migration

- Redo the tax computation for efficiency and reliability
- Add test suite for checking obtained JSON for different tax configurations
- SII description computation reworked for new computed writable
- Added migration scripts

pedrobaeza committed:
[REN] l10n_es_aeat_sii_oca: Rename from l10n_es_aeat_sii

For avoiding collapse on Odoo Apps Store.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca: Avoid KeyError on taxes with no amount

On taxes with no amount ("IVA soportado exento" for example), the key `amount` is not
filled, provoking a problem sending it to SII.

We avoid it initializing full expected dictionary structure on first occurrence.

This commit also contains a little fix for avoiding the crash of `merge_dict` method
not having "TipoImpositivo" key in the dictionaries.

brendapaniagua committed:
Translated using Weblate (Catalan)

Currently translated at 50.5% (105 of 208 strings)

Translation: l10n-spain-13.0/l10n-spain-13.0-l10n_es_aeat_sii_oca
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-13-0/l10n-spain-13-0-l10n_es_aeat_sii_oca/ca/

brendapaniagua committed:
Translated using Weblate (Catalan)

Currently translated at 58.7% (122 of 208 strings)

Translation: l10n-spain-13.0/l10n-spain-13.0-l10n_es_aeat_sii_oca
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-13-0/l10n-spain-13-0-l10n_es_aeat_sii_oca/ca/

HaraldPanten committed:
[FIX]+[IMP]l10n_es_aeat_sii_oca: Added new description image and fixed a micro typo in manifest file

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca: Proper registration key domain

Closes #1522

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca: Change key when invoicing purchase order

Steps to reproduce:

1. Create vendor with fiscal position "Régimen intracomunitario".
2. Assign in the fiscal position the SII registration key for purchases
   "[09]".
3. Create a purchase order for that vendor and confirm it.
4. Press on "Create Bill" button.

Current behavior:

The vendor bill is created with SII registration key "[01]".

Expected behavior:

The vendor bill is created with SII registration key "[09]".

This is provoked by the sequence Odoo follows for creating the vendor
bill, using a combination of default values + the triggering of
onchanges. The problem is that the onchange of the fiscal position is
triggered once at the beginning (when no partner nor fiscal position
is set), and not triggered again when this happens.

The solution is to catch on the partner onchange also the possible
condition for changing the SII registration key, and trigger it
manually.

Unfortunately, a regression test can't be added, as this module doesn't
depend on purchase for recreating the steps above.

Fixes ##1524

pedrobaeza committed:
[FIX+IMP] l10n_es_aeat_sii_oca: Incorrect sign in not_amount_in_total + extra tests

On customer invoices, the amount not included in total was summing
instead of substracting, and as the result comes already reversed, this
deviates the correct result.

Extra tests has been added for testing this and also check if here the
problem at #1552 was reproduced in this version, which isn't.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca: Proper ImporteTotal value

Since 01/01/2020, this value should include the deductible part of
intra-community, extra-community and other operations with double
tax imposition that both nullifies in the final invoice amount.

We handle this adding a new tax mapping key for substracting instead
of adding tax amounts in the amount to correct in ImporteTotal, as the
sign comes reversed for these taxes.

Fixes #1490

joao-p-marques committed:
[IMP] l10n_es_aeat_sii_oca: black, isort, prettier

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca: Exclude base amount from ImporteTotal

If you use "IVA Exento Repercutido No Sujeto", its base amount
shouldn't be included in `ImporteTotal`.

We introduce here a mechanism to declare taxes for which base shouldn't
be included, and define such tax in it.

joao-p-marques committed:
[MIG] l10n_es_aeat_sii_oca: Migration to 14.0

Also apply the part of https://github.com/OCA/l10n-spain/commit/86ff0f8ce82250618061b6660bc0d394439d0c7d that related to this.

TT26438

pedrobaeza committed:
[REF+IMP] l10n_es_aeat/l10n_es_aeat_sii_oca: Re-arrange certificate tests

- Put minimal certificate test in l10n_es_aeat
- Remove deprecation warning due to use of password as str.
- Extend properly tests in l10n_es_aeat_sii_oca for the SII part.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: Include "Importación bienes de inversión" in SFRS

If not, DUA invoices for this type of good can't be sent.

joao-p-marques committed:
[FIX] l10n_es_aeat_sii_oca: Fix move_type reference

Step left from migration

TT28423

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca: Propert journal type on invoices

There's now a check about the journal type with the move type, so we
can't change types without changing the journal.

pedrobaeza committed:
Update ROADMAP.rst

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca: Hook on post low level method

We were intercepting the high level method for posting the invoice,
but this method calls a low level one that is the one that is always
called.

There's a chance of having the high level method to be bypassed and
not called, so we need to intercept the low level one.

We also need to filter the non invoices records now at this level.

pedrobaeza committed:
[FIX+REF] l10n_es_aeat_sii_oca: Remove offending test + Move json files to subdirectory

The test was incorrect, as the invoice is already posted, and now
there's not too much sense to test such regression.

Moved so that they are not listed at the beginning of the test folder,
and thus it's easier to locate test files.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca: Don't assign specific refund type

Adaptation of cc0e79d1363e61dc05c3d8959f4aac866d8d6b02

etobella committed:
[IMP] l10n_es_aeat_sii_oca: remove PyOpenSSL dependancy

joao-p-marques committed:
[FIX] l10n_es_aeat_sii_oca: SII Refund Type not required

Make the SII Refund Type not required nor available for account.move records that are not "in_invoices" or "out_invoices"

TT30339

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca: Don't make required + visible location info on purchases

The SII keys for requiring to input location information are for sales,
not for purchases.

Fixes #1626

Commit:
[14.0][FIX] l10n_es_aeat_sii_oca: type in domain to move_type

victoralmau committed:
[IMP] l10n_es_aeat_sii_oca: Allow send invoices with LC type (create _get_sii_invoice_type() function to set 'TipoFactura'

HaraldPanten committed:
[UPD] ROADMAP.rst

pedrobaeza committed:
[IMP] l10n_es_aeat_sii: Add 24% retentions in NotIncludedInTotal

Since https://github.com/odoo/odoo/pull/68924, there are 2 new retention
taxes that were included in total by omission, so we need to explicitly
remove them.

victoralmau committed:
[IMP] l10n_es_aeat_sii_oca: Add SII third-party
TT30681

jesusVMayor committed:
[FIX]l10n_es_aeat_sii_oca: Fix field name on default type function.

victoralmau committed:
[FIX] l10n_es_aeat_sii_oca: Only assign sii_registration_key if is set in fiscal position

acysos committed:
[ADD] l10n_es_aeat_sii_oca: add Araba/Alava and Bizkaia/Vizvaya

omar7r committed:
[FIX] l10n_es_aeat_sii_oca: Fix obtaining certificate in multicompany environment

omar7r committed:
[FIX] l10n_es_aeat_sii_oca: Send invoice's commercial_partner_id

omar7r committed:
[FIX] l10n_es_aeat_sii_oca: hook for obtaining the partner to send SII

victoralmau committed:
[IMP] l10n_es_aeat_sii_oca: Rename thirdparty fields to allow inheritance in other addons.
Hide thirdparty fields in some cases:
- Hide thirdparty fields in moves when it is not an invoice.
- Hide thirdparty field in journals when it is not sale/purchase type.
- Avoid modifying third-party number when an invoice has already been sent to the SII.
- Avoid modifying customer invoice name when an invoice has already been sent to the SII.
TT32612

pedrobaeza committed:
[OU-FIX] l10n_es_aeat_sii_oca: `folder` field required

We fill it with a wildcard text for complying with the requirement, but
its content is not so important, specially if the certificate is
already in use, as this one is only used for composing the new decrypted
certificates file path, not existing ones.

ValentinVinagre committed:
[MIG] l10n_es_aeat_sii_oca: Migration to 15.0

etobella committed:
[IMP] l10n_es_aeat_sii_oca: Allow to pass a partner without ES at the beginning of VAT

etobella committed:
[IMP] l10n_es_aeat_sii_oca: Check if the sii is enabled checking all the companies

igallart committed:
[IMP] l10n_es_aeat_sii: Se añade nuevo mapeo para el SII

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca: Adjust tests according addition of s_iva0_ns

victoralmau committed:
[IMP] l10n_es_aeat_sii_oca: Remove required condition in reference field from invoices
TT32612

pedrobaeza committed:
[FIX] black formatting

LoisRForgeFlow committed:
[IMP] l10n_es_aeat_sii_oca: migrate to common aeat.tax.agency

* Remove duality with ir.config.parameters. Now it is required to
  set a tax agency.
* migration scripts included.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca: Avoid access error to SII taxes map

Steps to reproduce the problem:

- Use a user with no AEAT permission.
- Create a vendor bill.
- Select a supplier with "Importación con DUA" fiscal position.

Current behavior: You get a permission error on 'SII - Mapeo de impuestos'

Expected behavior: No error

We need to do the operation with sudo. It was included, but later once
the search is done, which was too late.

TT35508

LoisRForgeFlow committed:
[FIX] l10n_es_aeat_sii_oca: actualizar url de WSDLs

Los links antiguos se han discontinuado y no se están actualizando.

Esto a afectado a la conexión a los entornos de pruebas ya que el
pasado 21 de marzo se cambiaron las url de los entornos de pre-
producción y esto no se vio reflejado en los WSDL de links anteriores.

pere-aquarian committed:
Translated using Weblate (Catalan)

Currently translated at 97.9% (191 of 195 strings)

Translation: l10n-spain-15.0/l10n-spain-15.0-l10n_es_aeat_sii_oca
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-15-0/l10n-spain-15-0-l10n_es_aeat_sii_oca/ca/

pere-aquarian committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (195 of 195 strings)

Translation: l10n-spain-15.0/l10n-spain-15.0-l10n_es_aeat_sii_oca
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-15-0/l10n-spain-15-0-l10n_es_aeat_sii_oca/es/

Noel000 committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (195 of 195 strings)

Translation: l10n-spain-15.0/l10n-spain-15.0-l10n_es_aeat_sii_oca
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-15-0/l10n-spain-15-0-l10n_es_aeat_sii_oca/ca/

etobella committed:
[FIX] l10n_es_aeat_sii_oca: Use partner hook

We should use _sii_get_partner function as someone can use the hook

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca: EU VAT number requires country code

Using l10n_es_aeat `_parse_aeat_vat_info`, the VAT number for European
is returned without the country code, but we need to send it when
informing a document type 02 to the SII.

TT36220

colmeirin committed:
[FIX] l10n_es_aeat_sii_oca: Don't cut VAT type 3 identifiers

Fixes #2248 vat error with exportación and Canarias.

LoisRForgeFlow committed:
[IMP] l10n_es_aeat_sii_oca: finally drop sii_tax_agency_id field.

jabelchi committed:
Translated using Weblate (Catalan)

Currently translated at 99.4% (191 of 192 strings)

Translation: l10n-spain-15.0/l10n-spain-15.0-l10n_es_aeat_sii_oca
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-15-0/l10n-spain-15-0-l10n_es_aeat_sii_oca/ca/

flachica committed:
[OU-FIX] l10n_es_aeat_sii_oca: Error tipográfico en una línea de código (#2305)

pedrobaeza committed:
[IMP] l10n_es_aeat_sii_oca: Avoid deprecation warning in Environment.manage()

pedrobaeza committed:
[IMP] l10n_es_aeat_sii_oca: Include new tax 5%

Following https://github.com/odoo/odoo/pull/95967, the mapping is done
in the SII corresponding files.

TT38004

dalonsod committed:
[FIX] l10n_es_aeat_sii_oca: admit E3 export exempt cause

* Before this fix, when an invoice fiscal position is marked as Export, SII
  exempt cause is currently forced to E2.
* E3 exempt cause should also be considered in this situation. This commit fix
  it. E2 is still selected as default in other cases, anyway.

yajo committed:
[FIX] l10n_es_*: remove pyOpenSSL and cryptography dependencies

- pyOpenSSL was replaced by cryptography and is no longer used.
- cryptography has a specific version dependency specified in `l10n_es_aeat`, which in turn is dependency of all these modules. So there's no real need to indicate that dependency in these files.

@moduon MT-1067

omar7r committed:
[IMP] l10n_es_aeat_sii_oca: Include ISP investments taxes and 'BienInversion' mark

alvaro-domatix committed:
[FIX] l10n_es_aeat_sii_oca: EU VAT number requires country code

etobella committed:
[IMP] l10n_es_aeat + l10n_aeat_sii_oca: Add standard function for getting taxes

- New hook for managing changes on the function
- New field for tax_amount
- Call l10n_es_aeat method in SII instead of own (+ remove specific one)

Shide committed:
[FIX] l10n_es_aeat_sii_oca: Remove `thirdparty_invoice` field from l10n_es_aeat_sii_oca

victoralmau committed:
[FIX] l10n_es_aeat_sii_oca: Remove thirdparty_invoice + thirdparty_number fields from l10n_es_aeat_sii_oca

dalonsod committed:
[FIX] l10n_es_aeat_sii_oca: Remove ISP taxes from NotIncludedInTotalNegative

Including l10n_es.account_tax_template_p_iva*_isp taxes in
NotIncludedInTotalNegative map section actually adds tax amount
to ImporteTotal. Removing them ImporteTotal becomes BaseImponible,
that is the expected behavior according AEAT.

Fixes #2669

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca: Proper places for some food taxes reduction

Refinement of #2836

- Missing tax "0% IVA soportado (servicios corrientes)"
- Bad tax group for taxes "Importaciones servicios corrientes"
- Missing "Recargo equivalencia 0%" taxes
- Missing new taxes in NotIncludeInTotalNegative

ramiadavid committed:
[FIX] l10n_es_aeat_sii_oca: compute macrodata from company currency amount

Shide committed:
[IMP] l10n_es_aeat_sii_oca: Includes new taxes 0%, 0.62% and 5%

Commit:
[IMP] l10n_es_aeat_sii_oca: pre-commit stuff

Commit:
[MIG] l10n_es_aeat_sii_oca: Migration to 16.0

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca: Missing retention in species taxes

Commit:
[UPD] Update l10n_es_aeat_sii_oca.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-16.0/l10n-spain-16.0-l10n_es_aeat_sii_oca
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-16-0/l10n-spain-16-0-l10n_es_aeat_sii_oca/

polqubiq committed:
[FIX] l10n_es_aeat_sii_oca: fix error sii header

If the company VAT doesn't contain `ES`, the SII sending fails,
as it was hardcoded to strip the first 2 characters.

Now it uses the standard AEAT function.

OCA-git-bot committed:
l10n_es_aeat_sii_oca 16.0.1.0.1

pedrobaeza committed:
[FIX+IMP] l10n_es_aeat*: VAT prorate proper encapsulation + names + tests

polqubiq committed:
[MIG] l10n_es_aeat_sii_oca: informative wizard invoices sii

Commit:
[UPD] Update l10n_es_aeat_sii_oca.pot

OCA-git-bot committed:
l10n_es_aeat_sii_oca 16.0.1.1.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-16.0/l10n-spain-16.0-l10n_es_aeat_sii_oca
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-16-0/l10n-spain-16-0-l10n_es_aeat_sii_oca/

OCA-git-bot committed:
l10n_es_aeat_sii_oca 16.0.1.2.0

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 6.5% (13 of 199 strings)

Translation: l10n-spain-16.0/l10n-spain-16.0-l10n_es_aeat_sii_oca
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-16-0/l10n-spain-16-0-l10n_es_aeat_sii_oca/sl/

yajo committed:
[FIX] l10n_es_aeat_sii_oca: cliente intracomunitario, entrega extracomunitaria

Una venta a un cliente intracomunitario enviada al extranjero es exportación.

Nunca se puede informar el tipo de identificador 02 para estos envíos porque la AEAT los rechaza siempre. Al poner identificador 06, funciona bien.

Fix https://github.com/OCA/l10n-spain/issues/2969

@moduon MT-2407

Co-authored-by: Rafael Blasco <8736623+rafaelbn@users.noreply.github.com>

pedrobaeza committed:
[IMP] l10n_es_aeat_sii_oca: Change registration key to computed writable

In this version, there's no onchange mitigation, but other similar
quirks, so let's convert the same as in previous versions the field
to computed writable to reduce code and use more modern approach.

pedrobaeza committed:
[IMP] l10n_es_aeat_sii_oca: Convert sii_refund_type to computed writable

Saving a lot of code and making it more predictable.

Commit:
[UPD] Update l10n_es_aeat_sii_oca.pot

OCA-git-bot committed:
l10n_es_aeat_sii_oca 16.0.1.2.1

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-16.0/l10n-spain-16.0-l10n_es_aeat_sii_oca
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-16-0/l10n-spain-16-0-l10n_es_aeat_sii_oca/

pedrobaeza committed:
Revert "[IMP] l10n_es_aeat_sii: Se añade nuevo mapeo para el SII"

This reverts commit d6a5cb83223193e4ac87cb3e2d6b9029d38ef96c.

This provokes an inconsistency, sending total amount not counting
it, but being on the tax detail. Morover, this tax is intended
for not being reported (that's why it says "No sujeto").

OCA-git-bot committed:
[UPD] README.rst

pedrobaeza committed:
Revert "[FIX] l10n_es_aeat_sii_oca: Adjust tests according addition of s_iva0_ns"

This reverts commit 424fcc821973ca59f42adf32eed0f724b0b66016.

OCA-git-bot committed:
l10n_es_aeat_sii_oca 16.0.1.2.3

Commit:
[UPD] Update l10n_es_aeat_sii_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-16.0/l10n-spain-16.0-l10n_es_aeat_sii_oca
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-16-0/l10n-spain-16-0-l10n_es_aeat_sii_oca/

Commit:
[REF] account.move extraemos parte de sii a un mixin.
[ADD] módulo l10n_es_pos_sii para permitir enviar pedidos del TPV al sii

Roadmap:
 - Comprobar que funcionalidad de account.move sigue siendo correcta.
 - Posibilidad de enviar al sii una sesión (tipo summary)
 - Envío devoluciones al SII -> POS
 - Cancelar envío al SII -> POS

Yadier-Tecnativa committed:
[FIX] l10n_es_aeat_sii_oca: broken view in Aeat SII Map Form

OCA-git-bot committed:
l10n_es_aeat_sii_oca 16.0.1.3.0

ljsalvatierra-factorlibre committed:
[IMP] l10n_es_aeat_sii_oca: move reversal defaul sii_refund_type

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
l10n_es_aeat_sii_oca 16.0.1.3.1

victoralmau committed:
[FIX] l10n_es_aeat_sii_oca: Proper forbidden unlinks

Related to https://github.com/OCA/l10n-spain/commit/f635bcafabbb15aaa90b6c5699e77fb44aad45df

TT44504

pedrobaeza committed:
[IMP] l10n_es_aeat_oca_sii: Add in ROADMAP "Devolución IVA viajeros"

OCA-git-bot committed:
[UPD] README.rst

ljsalvatierra-factorlibre committed:
[FIX] l10n_es_aeat_sii_oca: rename method confirm_one_invoice
forward port #3165

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (199 of 199 strings)

Translation: l10n-spain-16.0/l10n-spain-16.0-l10n_es_aeat_sii_oca
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-16-0/l10n-spain-16-0-l10n_es_aeat_sii_oca/es/

OCA-git-bot committed:
[UPD] README.rst

Commit:
[UPD] Update l10n_es_aeat_sii_oca.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-16.0/l10n-spain-16.0-l10n_es_aeat_sii_oca
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-16-0/l10n-spain-16-0-l10n_es_aeat_sii_oca/

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca: Avoid invoice copy with type change in tests

There's a new integrity check for avoiding inconsistent states. We were
using a trick on tests for quickly copying a customer invoice,
and converting the copy to vendor bill, but data got inconsistent doing
it.

For what we were testing, that was not a problem, but with the new
check, the operation is aborted.

Thus, we create the vendor bill from scratch, refactoring a method for
doing it easily.

We also take the occasion to improve a bit the number of invoice copies
done.

OCA-git-bot committed:
oca-github-bot post-merge updates

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (201 of 201 strings)

Translation: l10n-spain-16.0/l10n-spain-16.0-l10n_es_aeat_sii_oca
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-16-0/l10n-spain-16-0-l10n_es_aeat_sii_oca/es/

dalonsod committed:
[FIX] l10n_es_aeat_sii_oca: wrong job channel name

Starting from 14.0, root. prefix for channel name is no longer required

Fixes #3147

OCA-git-bot committed:
[BOT] post-merge updates

Shide committed:
[FIX] l10n_es_aeat_sii_oca: Check for SII Exceptions when posting an invoice

OCA-git-bot committed:
[BOT] post-merge updates

aritzolea committed:
[IMP] l10n_es_aeat_sii_oca: Add neutralize.sql file for test enabling

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[FIX] l10n_es_aeat_sii_oca: Inherit proper view

OCA-git-bot committed:
[BOT] post-merge updates

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca: Avoid singleton error on invoice posting

If you post more than one invoice at a time (using the action in the
list), and one of them is already sent to SII, there's a singleton error
because a typo in the code using `self` instead of `invoice`.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca: Crash on sending cancellation to SII

When sending a cancellation to SII, `FacturaExpedida` element in the
data structure is not built and sent, so current override (after the
mixin refactoring) is not taking into account this, crashing on
modifying a missing element.

We avoid the problem don't modifying such elements if cancelling.

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

dalonsod committed:
[IMP] l10n_es_aeat_sii_oca: show account registration date

In v13 this field was removed from SII result tab, while merging invoice
and vendor bill forms.

OCA-git-bot committed:
[BOT] post-merge updates

aritzolea committed:
[FIX] l10n_es_*: Fix minimum models read permissions for readonly accounting group

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[IMP] Refactor para futura implementación Veri*FACTU.

Creación de un nuevo modelo en `l10n_es_aeat` llamado *aeat_mixin*, del
que heredan *sii.mixin* (ya existente) y un nuevo *verifactu.mixin*.
Refactor de *sii.mixin* pasando funcionalidad a *aeat_mixin*
Se han renombrado varios campos sii_* a aeat_* ya que serán utilizados
por ambos.

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca: Exclude tax agency connection from tests

Since several days ago, the agency is returning tiemout when trying
to connect to it. It's not happening for all the hosts, but GitHub
is affected.

The only solution for now is to disable the connection tests.

We also remove the old Gipuzkoa agency.

Fixes #3589

OCA-git-bot committed:
[BOT] post-merge updates

pedrobaeza committed:
[IMP] l10n_es_aeat_sii_oca: Reactivate agencies tests

It seems they are working regularly again.

OCA-git-bot committed:
[BOT] post-merge updates

mide-dynapps committed:
[FIX] l10n_es_aeat_sii_oca: check coming from move

Some others modules call reversal but active_ids is not an account.move: module OE helpdesk_account (from a helpdesk ticket, we can create a "Refund")

OCA-git-bot committed:
[BOT] post-merge updates

acysos committed:
[IMP] l10n_es_aeat_sii_oca: send with quarters only for voluntary

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca/l10n_es_intrastat_report: Rename test helper method

Since https://github.com/odoo/odoo/commit/07be14e3086f473ed9a8c4b12ea7c,
Odoo has a helper method in AccountTestInvoicingCommon called
`_create_invoice`.

There are some tests here in this repository using the same name for
methods with the decorator `classmethod` and also inheriting from the
previous one, so they collide.

- l10n_es_aeat_sii_oca: l10n_es_pos_sii adds the inheritance on
  AccountTestInvoicingCommon
- l10n_es_intrastat_report: Direct inheritance.

The solution is to rename these methods to another name.

OCA-git-bot committed:
[BOT] post-merge updates

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca: Don't requeue pending invoices

Steps to reproduce:

- Enable SII in a company with connector with a delay of 48 hours.
- Create and validate an invoice. A job is configured for being executed
  in 48 hours.
- Next day, filter "Not sent to SII", and do Action > Send to SII.

Result: the pending invoices are requeued and the delay is reapplied,
so the invoices will be sent in 48 hours from that moment. Doing this
indefinitely, the invoices will never be sent!

Although it's a weird behavior, let's avoid it not requeuing the
invoices with pending jobs.

For this, a little refactor in the code has been done, not calling
_cancel_sii_jobs, but filtering another way, and not removing previous
failed jobs, which can be considered the only functional change, and
it's not bad IMO.

TT50641

Commit:
[UPD] Update l10n_es_aeat_sii_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-16.0/l10n-spain-16.0-l10n_es_aeat_sii_oca
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-16-0/l10n-spain-16-0-l10n_es_aeat_sii_oca/

Commit:
[UPD] Update l10n_es_aeat_sii_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-16.0/l10n-spain-16.0-l10n_es_aeat_sii_oca
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-16-0/l10n-spain-16-0-l10n_es_aeat_sii_oca/

manuelregidor committed:
[IMP] l10n_es_aeat_sii_oca: black, isort, prettier

manuelregidor committed:
[MIG]l10n_es_aeat_sii_oca: Migration to 17.0

pedrobaeza committed:
[IMP] l10n_es_aeat_sii_oca: Updated taxes october 2024

Following odoo/odoo@fbc72144b8a5ee85d835009a4413ce199fa19ac, this commit
puts the new taxes on the corresponding SII maps for being able to
send invoices containing them to the AEAT service.

TT51016

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca: Take into account inactive tax templates

Similar to #3744

TT51159

manuelregidor committed:
[IMP]l10n_es_aeat_sii_oca: sii con DUA

victoralmau committed:
[IMP] l10n_es_aeat_sii_oca: Add Enable SII to journals to enable compatibilities (hr_expense for example)

Steps to reproduce:
- Install hr_expense
- Create a purchase type journal named Expenses and uncheck the Enable SII field.
- Go to Expense > Configuration > Settings and define in Employee Expense Journal the
Expenses journal previously created
- Create an expense and an expense sheet
- Click on the Submit to manager button
- Click on the Approve button
- Click on the Post journal entries button
- The journal entry will be created correctly without any error

TT51825

victoralmau committed:
[IMP] l10n_es_aeat_sii_oca: Improve SII not sent filter to use sii_enabled field

TT52143

AaronHForgeFlow committed:
[MIG] l10n_es_aeat_sii_oca: Migration to v18

Commit:
[UPD] Update l10n_es_aeat_sii_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

AaronHForgeFlow commented:
Comenzaré de cero después de los muchos cambios que han habido en la 17.0

pedrobaeza commented:
Aarón, espera un poco más, que voy a hacer algunos cambios. Y luego nos reunimos para ver cómo atacar lo de los certificados. Luego te contacto por Matrix.

pedrobaeza committed:
[FIX] l10n_es_*: Fix intra/extra-community taxes refs and new exempt taxes

Odoo screwed up the tax XML-IDs in the middle of the 17.0 lifecycle,
introducing an unfolding of the exempt taxes, using existing ones
that belongs to the sales intra and extra community taxes.

After that, they have renamed the XML-IDs of both taxes to new ones.

We need to handle this in 2 ways:

- Detect and correct existing installations previous to this
  taxpocalypse for renaming them.
- Change the reference in all places to the new XML-IDs.

Besides, we have to add the unfolded exempt taxes in the reports.

OCA-git-bot committed:
[BOT] post-merge updates

AaronHForgeFlow committed:
[IMP] l10n_es_aeat_sii_oca: pre-commit auto fixes

sergio-teruel committed:
[IMP] l10n_es_aeat_sii_oca: Add methods to get tax agency from other sources like as journal

AaronHForgeFlow commented:
Probado con certificado en la web de la AEAT para pruebas, tanto con facturas de cliente como de compra e incluyendo cancelación. Ready to review.

acysos commented:
/ocabot migration l10n_es_aeat_sii_oca

pedrobaeza commented:
A tener en cuenta #4202

pedrobaeza committed:
[FIX] l10n_es_aeat_sii_oca: No taxable reason for services

When using the not subjected tax for services, the reason stated in
the journal was ignored, always putting "ImporteTAIReglasLocalizacion",
on contrary than with goods, so let's call the same method to get
the reason.

pedrobaeza commented:
Hay que incluir #4254

HaraldPanten commented:
@AaronHForgeFlow @pedrobaeza Cómo va avanzando el SII? Revisasteis lo que comentasteis de los certificados? Por saber si puedo empezar con las validaciones del PR.

Gracias.

AaronHForgeFlow commented:
Hola @HaraldPanten  Lo miramos en los OCA Days y como no estaba nada claro que usar los nuevos ceritificados fuera mejor, ni si Odoo introducirá nuevos cambios en nuevas versiones de momento lo dejamos tal cuál. Siempre se puede hacer otra PR como mejora más adelante.

También hicimos un par de pruebas funcionales en modo test y fue bien. Así que lo podéis revisar :)

HaraldPanten commented:
Perfecto! Pues miraré de atacarlo a partir de las próximas semanas 👍

pedrobaeza commented on l10n_es_aeat_sii_oca/views/account_move_views.xml:
```diff
@@ -42,75 +42,70 @@
-                            required="sii_enabled and move_type in ['out_refund','in_refund']"
-                            invisible="move_type not in ['out_refund','in_refund']"
-                        />
+                        name="thirdparty_number"
```
Todos estos cambios de indentación no parecen correctos. ¿Puedes revisar?

AaronHForgeFlow replied:
Lo miro, intencionados no son, quizás sea el pre-commit

AaronHForgeFlow replied:
Se había colado un '/>' extra. No entiendo como no petaba :confused:

pedrobaeza reviewed (commented):
¿Puedes poner el commit de la migración el último?

pedrobaeza reviewed (approved):
/ocabot merge nobump
